editoast/editoast_authz/src/regulator.rs

```diff
@@ -635,6 +703,30 @@ impl<S: StorageDriver> Regulator<S> {
         Ok(users)
     }
 
+    /// Get IDS of infras a subject can read
+    pub async fn list_authorized_infra(
+        &self,
+        user: &User,
+    ) -> Result<Authorization<Vec<i64>>, Error<S::Error>> {
```

Comment:
```suggestion
    ) -> Result<Authorization<Vec<Infra>>, Error<S::Error>> {
```

you can use `authz::model` types in return position as well (avoids the iteration in the end)

Reply:
done